src/FluentValidation/Validators/EmptyValidator.cs

```diff
@@ -32,20 +32,7 @@ public class EmptyValidator : PropertyValidator, IEmptyValidator {
 		}
 
 		protected override bool IsValid(PropertyValidatorContext context) {
-			switch (context.PropertyValue) {
-				case null:
-				case string s when string.IsNullOrWhiteSpace(s):
-				case ICollection c when c.Count == 0:
-				case Array a when a.Length == 0:
-				case IEnumerable e when !e.Cast<object>().Any():
-					return true;
-			}
-
-			if (Equals(context.PropertyValue, _defaultValueForType)) {
-				return true;
-			}
-
-			return false;
+			return NotEmptyValidator.IsEmpty(context.PropertyValue, _defaultValueForType);

```

Comment:
Nice to get rid of the considerable amount of duplication, although I'd recommend extracting the "IsEmpty" logic to an extension method instead. That would allow both validators to tap into the logic without cross-referencing each other which might be seen as a design smell.

Reply:
It's an internal implementation detail so I don't really think it matters here. Extension methods are nice for the public-facing API, but for the internals I don't think it's really needed.